### ISSUE TITLE
Field filter's timestamp UTC conversion does not eliminate BigQuery's partitions

### ISSUE THREAD

**Issue by SylvainGravejat** [metabase/metabase#44814]
### Describe the bug

Environment: BigQuery table, partitioned on a `created_at` timestamp column, and partition filter is set to be required.

The metabase question has a field filter on the `created_at`. The generated query results in the BigQuery error:

```
Cannot query over table without a filter over column(s) 'created_at' that can be used for partition elimination
```

Looking at the SQL that metabase generates, I see the field filter is applied this way:

```
TIMESTAMP_TRUNC(created_at, minute, 'UTC') BETWEEN ...
```

If I remove the 'UTC' conversion part, the query works:

```
TIMESTAMP_TRUNC(created_at, minute) BETWEEN ...
```

Note that the `created_at` field is already a UTC timestamp. AFAIK timestamps in BigQuery do not store the time zone information. So there is no need to apply a UTC conversion. I think it would make sense to drop that 'UTC' conversion when the field filter is on a BigQuery timestamp?

Also, here the problem is directly visible because the partition filter is set to be required, so we get an error. If it were not required, the query would work, but I'm guessing there would be no partition pruning, the whole table would be read, which is not good.

### To Reproduce

1. Have a BigQuery table with a partition on a timestamp column. Have the "partition filter = required" setting on that table.
2. In metabase, build a query on that table with a field filter on the timestamp column.
3. Apply a filter value and execute the query.
4. You should get the error mentioned: "Cannot query over table without a filter over column(s) 'xxx' that can be used for partition elimination"

### Expected behavior

I expect no BigQuery error to happen. (partition pruning would work).

### Logs

*No response*

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "fr-FR",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.23+9",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.23",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.23+9",
    "os.name": "Linux",
    "os.version": "5.15.0-1053-gke",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "bigquery-cloud-sdk"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "11.22"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.7.3"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2024-06-25",
      "tag": "v0.50.7",
      "hash": "431cd8f"
    },
    "settings": {
      "report-timezone": "UTC"
    }
  }
}
```

### Severity

Annoying

### Additional context

*No response*

**Comment by SylvainGravejat** [comment]
Nevermind, I solved it by changing the "report timezone" setting to "Database Default".

![image](https://github.com/metabase/metabase/assets/94012127/a39271e6-2c0f-4f97-af58-66574d33af0b)

*Possibly* an improvement could be : if that setting is UTC, then still don't apply the conversion (for a BigQuery timestamp column at least).

**Comment by calherries** [comment]
This could very well be QPD’s scope but it looks like an edge case of [Bigquery sync partitioned fields](https://github.com/metabase/metabase/pull/36892#top), so I assigned it to BEC.

**Comment by qnkhuat** [comment]
@calherries, I don't think it relates to the sync partitioned fields project at all. That projects only add the filter during sync. These filters are not automatically applied for questions, so it's probably a driver bug.

**Comment by calherries** [comment]
Ah okay, I didn't realise that was out of the project's scope. Then this is related to https://github.com/metabase/metabase/issues/37185

**Comment by snoe** [comment]
I believe that this may no longer be possible to hit from built queries.

Steps to reproduce:
```
(bigquery.tx/execute!
(format "CREATE TABLE %s (company STRING, founded TIMESTAMP)
                           PARTITION BY DATE(founded)
                           OPTIONS (require_partition_filter = TRUE);"
                          (fmt-table-name "partition_by_datetime")))

;; ensure the query in OP produces an error:
(-> {:query (format 
                                    "select * from %s where timestamp_trunc(founded,hour,'UTC') between '2022-01-01T00:01:00Z' and '2023-03-03T01:03:00Z'"
                                   (fmt-table-name "partition_by_datetime"))}
                           mt/native-query
                           qp/process-query
                           mt/rows)
```
Using the UI in native and MBQL queries create date filters trying to produce this kind of query. In every instance `between` gets desugared to `>= ....  < ....`  and properly uses the partition filter.


If anyone is still running into this I would appreciate knowing exactly what filters caused this error.

**Comment by SylvainGravejat** [comment]
On my end I don't have the error anymore (even when setting the metabase time zone to "Etc/UTC").

**Comment by paoliniluis** [comment]
@snoe have you changed the timezone of the query?

**Comment by snoe** [comment]
@paoliniluis yes, changing the timezone does put the timezone into `timestamp_trunc` but only on the RHS now. I can create a native query that exhibits the issue, but not from mbql/query builder/dashboard filters/native parameters.

I'm not sure if it was more consistent `:between` desugaring on our side or a change on bigquery's side that fixed this.
